tests/backup/backup_resiliency_test.go

```diff
@@ -1825,3 +1826,169 @@ var _ = Describe("{CancelAllRunningRestoreJobs}", func() {
+				err := CreateBackupWithValidation(ctx, backupName, SourceClusterName, bkpLocationName, backupLocationUID, appContextsToBackup, labelSelectors, BackupOrgID, srcClusterUid, "", "", "", "")
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Creation and Validation of backup [%s] of namespace (scheduled Context) [%s]", backupName, namespace))
+			}
+		})
+
+		Step("Taking backup of applications with incremental iteration of network delay in Milliseconds", func() {
+			log.InfoD("Taking backup of applications with incremental iteration of network delay in Milliseconds")
+			incrementalMilliseconds := 500
+			MinDelayInMilliseconds := 0
+			MaxDelayInMilliseconds := 100000
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			for _, namespace := range appNamespaces {
+				for i := MinDelayInMilliseconds; i <= MaxDelayInMilliseconds; i += incrementalMilliseconds {
+					log.Infof(fmt.Sprintf("Adding a delay of %dms  to the applications nodes", i))
+					err := InjectNetworkErrorInNodes(appNodesMap[namespace], "delay", "add", 0, i)
+					dash.VerifyFatal(err, nil, fmt.Sprintf("Adding a delay of %dms to nodes", i))
+					backupName := fmt.Sprintf("%s-%s-delay-%dms-%s", BackupNamePrefix, namespace, i, RandomString(5))
+					appContextsToBackup := FilterAppContextsByNamespace(scheduledAppContexts, []string{namespace})
+					err = CreateBackupWithValidation(ctx, backupName, SourceClusterName, bkpLocationName, backupLocationUID, appContextsToBackup, labelSelectors, BackupOrgID, srcClusterUid, "", "", "", "")
+					if err != nil {
+						log.Infof(fmt.Sprintf("The backup creation failed with error [%v]", err))
+						bkpStatus, bkpReason, err := CheckBackupStatus(backupName, ctx)
+						dash.VerifyFatal(err, nil, fmt.Sprintf("fetching backup status for backup [%s]", backupName))
+						if bkpStatus == api.BackupInfo_StatusInfo_Failed && strings.Contains(bkpReason, "Timed out while waiting for a status update.") {
+							log.InfoD(fmt.Sprintf("Backup [%s] failed when network delay of %dms injected to the applications node", backupName, i))
+							log.Infof(fmt.Sprintf("Deleting the delay of %dms  to the applications nodes", i))
+							err := InjectNetworkErrorInNodes(appNodesMap[namespace], "delay", "del", 0, i)
+							dash.VerifyFatal(err, nil, fmt.Sprintf("Removing delay of %dms from nodes when backup failed", i))
+							log.Infof(fmt.Sprintf("Create the backup when delay of %dms is removed and verify backup is succeded", i))
+							finalBackupName := fmt.Sprintf("%s-%s-nodelay-%s", BackupNamePrefix, namespace, RandomString(5))
+							err = CreateBackupWithValidation(ctx, finalBackupName, SourceClusterName, bkpLocationName, backupLocationUID, appContextsToBackup, labelSelectors, BackupOrgID, srcClusterUid, "", "", "", "")
+							dash.VerifyFatal(err, nil, fmt.Sprintf("Creation and Validation of backup [%s] of namespace (scheduled Context) [%s]", backupName, namespace))
```

Comment:
minor improvement in logging , "Creation and Validation of backup after removing the network delay"

Reply:
Done